pkg/server/registry/apigroups/acorn/images/sign.go

```diff
@@ -0,0 +1,129 @@
+package images
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/acorn-io/mink/pkg/stores"
+	"github.com/acorn-io/mink/pkg/types"
+	"github.com/acorn-io/mink/pkg/validator"
+	apiv1 "github.com/acorn-io/runtime/pkg/apis/api.acorn.io/v1"
+	acornsign "github.com/acorn-io/runtime/pkg/cosign"
+	"github.com/acorn-io/runtime/pkg/imagedetails"
+	"github.com/acorn-io/runtime/pkg/images"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	cremote "github.com/sigstore/cosign/v2/pkg/cosign/remote"
+	"github.com/sigstore/cosign/v2/pkg/oci/mutate"
+	ociremote "github.com/sigstore/cosign/v2/pkg/oci/remote"
+	"github.com/sigstore/cosign/v2/pkg/oci/static"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func NewImageSign(c client.WithWatch, transport http.RoundTripper) rest.Storage {
+	return stores.NewBuilder(c.Scheme(), &apiv1.ImageSignature{}).
+		WithValidateName(validator.NoValidation).
+		WithCreate(&ImageSignStrategy{
+			client:       c,
+			transportOpt: remote.WithTransport(transport),
+		}).Build()
+}
+
+type ImageSignStrategy struct {
+	client       client.WithWatch
+	transportOpt remote.Option
+}
+
+func (t *ImageSignStrategy) Create(ctx context.Context, obj types.Object) (types.Object, error) {
+	var (
+		isig = obj.(*apiv1.ImageSignature)
+		err  error
+	)
+
+	if isig.Name == "" {
+		ri, ok := request.RequestInfoFrom(ctx)
+		if ok {
+			isig.Name = ri.Name
+		}
+	}
+	ns, _ := request.NamespaceFrom(ctx)
+	opts := []remote.Option{t.transportOpt}
+	if isig.Auth != nil {
```

Comment:
Does `Auth` ever get set on this type? I don't think I saw that happen anywhere.

Reply:
Also I don't think we normally pass registry credentials from the client to the apiserver like this. If you look at most of our image operations, we use functions like `images.GetAuthenticationRemoteOptionsWithLocalAuth` to get the credentials.

Reply:
Yeah, it happens on line 60 :grimacing: 
Yeah I used to use `images.GetAuthenticationRemoteOptionsWithLocalAuth` elsewehere, but somehow I ended up copying this over from ImageDetails.
I may want to dive into the differences of those variations. :thinking:

Reply:
Nevermind, found my mistake :man_facepalming: